crates/uv/src/commands/project/add.rs

```diff
@@ -0,0 +1,87 @@
+use anyhow::Result;
+use std::str::FromStr;
+use uv_distribution::pyproject_mut::PyProjectTomlMut;
+
+use distribution_types::IndexLocations;
+use pep508_rs::Requirement;
+use pypi_types::LenientRequirement;
+use uv_cache::Cache;
+use uv_configuration::{ExtrasSpecification, PreviewMode, Upgrade};
+use uv_distribution::ProjectWorkspace;
+use uv_warnings::warn_user;
+
+use crate::commands::{project, ExitStatus};
+use crate::printer::Printer;
+
+/// Add one or more packages to the project requirements.
+#[allow(clippy::too_many_arguments)]
+pub(crate) async fn add(
+    requirements: Vec<String>,
+    preview: PreviewMode,
+    cache: &Cache,
+    printer: Printer,
+) -> Result<ExitStatus> {
+    if preview.is_disabled() {
+        warn_user!("`uv add` is experimental and may change without warning.");
+    }
+
+    // Find the project requirements.
+    let project = ProjectWorkspace::discover(&std::env::current_dir()?, None).await?;
+
+    let mut pyproject = PyProjectTomlMut::from_toml(project.current_project().pyproject_toml())?;
+    for req in requirements {
+        let req = Requirement::from(LenientRequirement::from_str(&req)?);
```

Comment:
I think we should just use `pep508_rs::Requirement::from_str` instead of `LenientRequirement`... We are a little inconsistent about it, but in general, `LenientRequirement` is really intended for third-party metadata over which the user has no control. We shouldn't let them add invalid requirements here.

Reply:
This also applies to the modification code in `crates/uv-distribution/src/pyproject_mut.rs`. What do you think?

Reply:
That makes sense to me, I removed the use of  `LenientRequirement`.